src/compile_cache.cc

```diff
@@ -283,23 +285,33 @@ void CompileCacheHandler::MaybeSave(CompileCacheEntry* entry,
   MaybeSaveImpl(entry, func, rejected);
 }
 
-// Layout of a cache file:
-// [uint32_t] magic number
-// [uint32_t] code size
-// [uint32_t] code hash
-// [uint32_t] cache size
-// [uint32_t] cache hash
-// .... compile cache content ....
-void CompileCacheHandler::Persist() {
+/**
+ * Persist the compile cache accumulated in memory to disk.
+ *
+ * To avoid race conditions, the cache file includes hashes of the original
+ * source code and the cache content. It's first written to a temporary file
+ * before being renamed to the target name.
+ *
+ * Layout of a cache file:
+ *   [uint32_t] magic number
+ *   [uint32_t] code size
+ *   [uint32_t] code hash
+ *   [uint32_t] cache size
+ *   [uint32_t] cache hash
+ *   .... compile cache content ....
+ */
+void CompileCacheHandler::Persist(bool keep_deserialized_cache) {
```

Comment:
Nit: Can we change the function signature to use enums? It's really hard to read a code like persist(true)

```suggestion
void CompileCacheHandler::Persist(Mode mode = Mode::STORE_CACHE) {
```

Reply:
Removed this option